addons/mail/static/src/messaging.js

```diff
@@ -494,4 +497,40 @@ export class Messaging {
     stopCall(threadId) {
         this.threads[threadId].inCall = false;
     }
+
+    // -------------------------------------------------------------------------
+    // drag & drop
+    // -------------------------------------------------------------------------
+
+    /**
+     * @param {DragEvent} ev 
+     */
+    _onDragEnter(ev) {
+        if (
+            this._dragCount === 0 &&
+            ev.dataTransfer?.types.includes("Files")
+        ) {
+            this.isDraggingFile = true;
+        }
+        this._dragCount++;
+    }
+
+    /**
+     * @param {DrageEvent} ev
+     */
+    _onDragLeave(ev) {
+        this._dragCount--;
+        if (this._dragCount === 0) {
+            this.isDraggingFile = false;
+        }
+    }
+
+    /**
+     * @param {DragEvent} ev
+     */
+    _onDrop(ev) {
+        this.isDraggingFile = false;
+        this._dragCount = 0;
+    }
```

Comment:
Better define a `FileDragDropzoneContainer` that is registered as a `main_component`.
As a `Component`, it can use `useExternalEventListener`.

I think you can take inspiration to `popover_service` for the design of `FileDragDropzone`:
Instead of a sub-component, it's a component hook `useFileDragDropzone(refName)`.

Components only have to `useFileDragDropzone(refName)` and it automatically show overlay styling on the flagged refs when dragging a file.